app/src/main/java/com/github/studydistractor/sdp/ui/DistractionListScreen.kt

```diff
@@ -0,0 +1,66 @@
+package com.github.studydistractor.sdp.ui
+
+import android.util.Log
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Divider
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import com.github.studydistractor.sdp.distraction.Distraction
+import com.github.studydistractor.sdp.distraction.DistractionViewModel
+import com.github.studydistractor.sdp.distraction.DistractionService
+
+/**
+ *  Screen showing the list of current available distraction on the DistractionService
+ */
+@Composable
+fun DistractionListScreen(distractionService : DistractionService,
+                          onClickingDistraction : () -> Unit = {},
+                          distractionViewModel : DistractionViewModel
+                      ) {
+    val distractions = distractionService.fetchDistractions()
+    LazyColumn(){
+        items(distractions) {distraction ->
+            DistractionLayout(distraction, onClickingDistraction, distractionViewModel)
+            Divider(
+                modifier = Modifier.padding(horizontal = 14.dp),
+                color = MaterialTheme.colors.onSurface.copy(alpha = 0.08f)
+            )
+        }
+    }
+}
+
+/**
+ * Distraction layout
+ */
+@Composable
+fun DistractionLayout(activity : Distraction,

```

Comment:
This should definitely be a separate file

Reply:
It is only used in the list, I don't understand why we should separate them, it is just a helper function to display things